packages/graphql-codegen-supermassive-schema-extraction-plugin/src/index.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  Types,
+  PluginValidateFn,
+  PluginFunction,
+} from "@graphql-codegen/plugin-helpers";
+import ts from "typescript";
+import { GraphQLSchema } from "graphql";
+import { printSchema, parse } from "graphql";
+import { extractImplicitTypesToTypescript } from "@graphitation/supermassive";
+import { extname } from "path";
+import { RawClientSideBasePluginConfig } from "@graphql-codegen/visitor-plugin-common";
+
+type PluginConfig = RawClientSideBasePluginConfig & {
+  files: string[];
+};
+
+export const plugin: PluginFunction<PluginConfig> = (schema: GraphQLSchema) => {
+  const tsContents: ts.SourceFile = extractImplicitTypesToTypescript(
+    parse(printSchema(schema)),
```

Comment:
Why do this?

Reply:
extractImplicitTypesToTypescript needs schemaAst not the object

Reply:
Doesn't this param, or one of the others, already contain the AST input? https://github.com/dotansimha/graphql-code-generator/blob/e3af1dbc80a998444a1d1b353f127f297c6a09d0/packages/utils/plugins-helpers/src/types.ts#L531

Reply:
Nope, I don't use the documents field. It's empty array in this case